sst/Assemble.cc

```diff
@@ -0,0 +1,185 @@
+#include "Assemble.hh"
+
+#ifdef SIMENG_ENABLE_SST_TESTS
+#include "llvm/MC/MCAsmBackend.h"
+#include "llvm/MC/MCAsmInfo.h"
+#include "llvm/MC/MCCodeEmitter.h"
+#include "llvm/MC/MCContext.h"
+#include "llvm/MC/MCInstrInfo.h"
+#include "llvm/MC/MCObjectFileInfo.h"
+#include "llvm/MC/MCObjectWriter.h"
+#include "llvm/MC/MCParser/MCAsmParser.h"
+#include "llvm/MC/MCParser/MCTargetAsmParser.h"
+#include "llvm/MC/MCRegisterInfo.h"
+#include "llvm/MC/MCStreamer.h"
+#include "llvm/Object/ELF.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+
+#if SIMENG_LLVM_VERSION < 14
+#include "llvm/Support/TargetRegistry.h"
+#else
+#include "llvm/MC/TargetRegistry.h"
+#endif
+
+#endif
+
+using namespace SST::SSTSimEng;
+
+#define ASSERT(expr, errStr)                                                 \
+  if (!(expr)) {                                                             \
+    std::cerr << "[SSTSimEngTest:Assembler] Error occured while assembling " \
+                 "source through LLVM:\n"                                    \
+              << errStr << std::endl;                                        \
+    exit(1);                                                                 \
+  }
+
+Assembler::Assembler(std::string source) {
+  std::string sourceWithTerminator = source + "\n.word 0";
+  assemble(source.c_str(), "aarch64");
+};
+
+Assembler::~Assembler(){};
+
+#ifdef SIMENG_ENABLE_SST_TESTS
```

Comment:
Because LLVM will on be injected as a dependency if `SIMENG_ENABLE_SST_TESTS` is defined. If this isn't done, then LLVM will always have to be a dependency for SST, even when we don't want to test, which is undesirable. 

`SIMENG_ENABLE_SST_TESTS` signals CMake to compile SST integration with LLVM

Reply:
Done.

Reply:
Looking at the comment, I don't think it captures the reasoning correctly. This being "conditional compilation so that LLVM is not a dependency when not using the testing framework"

Reply:
Comment improved, resolving